activesupport/lib/active_support/values/time_zone.rb

```diff
@@ -248,7 +249,18 @@ def [](arg)
       # A convenience method for returning a collection of TimeZone objects
       # for time zones in the USA.
       def us_zones
-        @us_zones ||= all.find_all { |z| z.name =~ /US|Arizona|Indiana|Hawaii|Alaska/ }
+        country_zones(:us)
+      end
+
+      # A convenience method for returning a collection of TimeZone objects
+      # for time zones in the country specified by its ISO 3166-1 Alpha2 code.
+      def country_zones(country_code)
+        code = country_code.to_s.upcase
+        @country_zones[code] ||=
+          TZInfo::Country.get(code).zone_identifiers.map do |tz_id|
```

Comment:
Why should it return a `ThreadSafe::Cache`? I've initialized cache instance earlier and only adding key/value pairs to it, never overwriting. Key is the country code, value is the array of timezone objects.
The another cache (`@lazy_zones_map`) in this file used similarly — created beforehand and added timezone objects by timezone names as keys afterwards when needed.